operator/internal/manifests/storage/configure.go

```diff
@@ -126,139 +126,69 @@ func ensureObjectStoreCredentials(p *corev1.PodSpec, opts Options) corev1.PodSpe
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAlibabaCloudSecretAccessKey,
-					},
-				},
-			},
+		return []corev1.EnvVar{
+			envVarFromSecret(EnvAlibabaCloudAccessKeyID, secretName, KeyAlibabaCloudAccessKeyID),
+			envVarFromSecret(EnvAlibabaCloudAccessKeySecret, secretName, KeyAlibabaCloudSecretAccessKey),
 		}
 	case lokiv1.ObjectStorageSecretAzure:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAzureStorageAccountName,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAzureStorageAccountName,
-					},
-				},
-			},
-			{
-				Name: EnvAzureStorageAccountKey,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAzureStorageAccountKey,
-					},
-				},
-			},
+		return []corev1.EnvVar{
+			envVarFromSecret(EnvAzureStorageAccountName, secretName, KeyAzureStorageAccountName),
+			envVarFromSecret(EnvAzureStorageAccountKey, secretName, KeyAzureStorageAccountKey),
 		}
 	case lokiv1.ObjectStorageSecretGCS:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name:  EnvGoogleApplicationCredentials,
-				Value: path.Join(secretDirectory, KeyGCPServiceAccountKeyFilename),
-			},
+		return []corev1.EnvVar{
+			envVarFromValue(EnvGoogleApplicationCredentials, path.Join(secretDirectory, KeyGCPServiceAccountKeyFilename)),
 		}
 	case lokiv1.ObjectStorageSecretS3:
-		storeEnvVars = []corev1.EnvVar{
-			{
-				Name: EnvAWSAccessKeyID,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAWSAccessKeyID,
-					},
-				},
-			},
-			{
-				Name: EnvAWSAccessKeySecret,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAWSAccessKeySecret,
-					},
-				},
-			},
+		storeEnvVars := []corev1.EnvVar{
+			envVarFromSecret(EnvAWSAccessKeyID, secretName, KeyAWSAccessKeyID),
+			envVarFromSecret(EnvAWSAccessKeySecret, secretName, KeyAWSAccessKeySecret),
 		}
-
 		if opts.S3 != nil && opts.S3.SSE.Type == SSEKMSType && opts.S3.SSE.KMSEncryptionContext != "" {
-			storeEnvVars = append(storeEnvVars, corev1.EnvVar{
-				Name: EnvAWSSseKmsEncryptionContext,
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: secretName,
-						},
-						Key: KeyAWSSseKmsEncryptionContext,
-					},
-				},
-			})
+			storeEnvVars = append(storeEnvVars, envVarFromSecret(EnvAWSSseKmsEncryptionContext, secretName, KeyAWSSseKmsEncryptionContext))
```

Comment:
This is missing in the STS variant. We may need to create two appending functions:
1. One pair as proposed here `envVarsWithWIF/envVarsNoWIF` that appends authentication env vars per store type.
2. One `envVarsServerSideEncryption` that appends env vars related to server side encryption per store type if applicable.

WDYT?